### DISCUSSION TITLE
How to use switch-on and set-timer function in one go

### DISCUSSION THREAD

**Discussion by vvmichielvv** [jasonacox/tinytuya#442] [General]
Hi Jason,

Thanks so much for all your efforts in keeping tuya controls up to date. I am a long time happy user of your code.

One question: I have a use case of switching on a heater for say 30mins and then automatically switch it off. I use a Tuya socket switch for that which works. However, it has sometimes trouble with its wifi connection. I want to absolutely make sure the socket switches off after 30 mins. If it hasn't switched on, then that is ok.

Is there any way in having ONE network call to the Tuya switch in which both the 'switch on' and 'set timer' DPs are set? That way, either nothing is switched (bad reception) or both on and off, but it never stays on. Is this with 'set_multiple_values' and is there a simple example that works?

regards, Michiel

**Comment by jasonacox** [comment]
Hi @vvmichielvv , thanks for the kind note! A huge credit to @uzlonewolf for the incredible protocol engineering and for the context of your request, the new function `set_multiple_values(dps)`.  

If you know the DP values you need to set for your device (perhaps [this](https://github.com/jasonacox/tinytuya?tab=readme-ov-file#version-33-devices)) you just create a dictionary of the DPS index and values. I believe something like this would work:

```python
# Set the DPS values
dps = {}
dps["1"] = True
dps["9"] = 1800

# Request the settings
set_multiple_values(dps)
```

Test it and let us know how it goes.

**Reply by vvmichielvv** [reply]
Hi Jason,

@uzlonewolf my thanks to you as well!

I have tried it and works like a charm! I also implemented it for dimmers (to always send both ON and dim value in one go, preventing a dimmer to switch on but failing to dim due to network connection problems in the second request), thanks for your help!

Regards, Michiel

**Reply by jasonacox** [reply]
Awesome! This is a great example use case for this function. I'm glad it work!
